hypothesis-python/src/hypothesis/strategies/_internal/numbers.py

```diff
@@ -425,38 +501,57 @@ def floats(
         elif is_negative(min_value):
             if is_negative(max_value):
                 return floats(
-                    min_value=-max_value, max_value=-min_value, width=width
+                    min_value=-max_value, max_value=-min_value, **strat_kw
                 ).map(operator.neg)
             else:
-                return floats(min_value=0.0, max_value=max_value, width=width) | floats(
-                    min_value=0.0, max_value=-min_value, width=width
+                return floats(min_value=0.0, max_value=max_value, **strat_kw) | floats(
+                    min_value=0.0, max_value=-min_value, **strat_kw
                 ).map(operator.neg)
-        elif count_between_floats(min_value, max_value) > 1000:
+        elif (
+            safe_count_between_floats(min_value, max_value, width, allow_subnormal)
+            > 1000
+        ):
```

Comment:
I'd be inclined to leave this as `elif count_between_floats(min_value, max_value) > 1000 or not allow_subnormal:` - handling subnormals in the next clause adds a huge pile of complexity which is very very rarely useful.